### DISCUSSION TITLE
Debugging (local) with Mac OS, CLion

### DISCUSSION THREAD

**Discussion by k-makarov** [ClickHouse/ClickHouse#32538] [Q&A] [answered]
Hello!

After several hours trying to stop the clickhouse client on breakpoint I gave up. 
I successfully build with Release and then with Debug configuration clickhouse-server and clickhouse-client, but can't get local debugger work on this project. Other projects debug fine tho. Maybe someone can help and investigate similar problem?

My configuration and other info:

MacOS (arm64)
CLion 2021.3 (Build #CL-213.5744.254, built on November 30, 2021)

Cmake Debug profile in CLion:

`Cmake options = -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER=/opt/homebrew/opt/llvm/bin/clang -DCMAKE_CXX_COMPILER=/opt/homebrew/opt/llvm/bin/clang++ -DCMAKE_AR=/opt/homebrew/opt/llvm/bin/llvm-ar -DCMAKE_RANLIB=/opt/homebrew/opt/llvm/bin/llvm-ranlib -DOBJCOPY_PATH=/opt/homebrew/opt/llvm/bin/llvm-objcopy`

<img width="992" alt="Снимок экрана 2021-12-11 в 00 08 30" src="https://user-images.githubusercontent.com/5754185/145641872-a37f1c8d-7783-4ab4-84c4-24e7823d6dfa.png">

Run/Debug configuration:

<img width="1049" alt="Снимок экрана 2021-12-11 в 00 13 04" src="https://user-images.githubusercontent.com/5754185/145642008-6807cd61-f9d3-4f92-a777-7b9b84c6755c.png">

After starting the local debug session (lldb) you can see my cout and output with exit code (server haven't been started - its ok) and breakpoint was ignored:

<img width="1728" alt="Снимок экрана 2021-12-11 в 00 21 17" src="https://user-images.githubusercontent.com/5754185/145642826-7b88bba1-f67b-4b03-bc28-dfc3bb7abfcd.png">

**Comment by alexey-milovidov** [comment, marked as answer]
FIrst you can check if breakpoints work from command line with `gdb` or `lldb`.
